addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -4,52 +4,116 @@ import "@website/snippets/s_website_form/000";  // force deps
+                        if (formButton) {
+                            formButton.disabled = false;
+                            formButton.classList.remove('disabled');
+                            formButton.querySelector('i.turnstile-spinner').remove();
                         }
-                    </script>
-                    <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
-                `).insertAfter('.s_website_form_send, .o_website_form_send');
-            }
-            return res;
-        },
-
-        /**
-         * Remove potential existing loaded script/token
-         */
-        cleanTurnstile: function () {
-            if (this.$('.s_turnstile').length) {
-                this.$('.s_turnstile').remove();
-            }
-        },
-
-        /**
-         * @override
-         * Discard all library changes to reset the state of the Html.
-         */
-        destroy: function () {
+                        if (webButton) {
+                            webButton.classList.remove('disabled');
+                            webButton.querySelector('i.turnstile-spinner').remove();
+                        }
+                    }
+                    function turnstileBecomeVisible() {
+                        this.wrapper.parentElement.style.display = ''
+                    }
+                </script>
+                <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
+            `);
+        }
+    },
+
+    /**
+     * Remove potential existing loaded script/token
+     */
+    cleanTurnstile: function () {
+        if (this.$('.s_turnstile').length) {
+            this.$('.s_turnstile').remove();
+        }
+    },
+
+    /**
+     * @override
+     * Discard all library changes to reset the state of the Html.
+     */
+    destroy: function () {
+        this.cleanTurnstile();
+        this._super(...arguments);
+    },
+};
+
+const signupTurnStile = {
+    ...turnStile,
+
+    addSpinner(button) {
+        let spinner = document.createElement('i')
+        spinner.classList.add('fa', 'fa-refresh', 'fa-spin', 'turnstile-spinner')
+        button.classList.add('disabled');
+        button.prepend(spinner)
+    },
+
+    async willStart() {
+        this._super(...arguments);
+        if(!session.turnstile_site_key) return;
+        let button = this.el.querySelector('button[type="submit"]')
+        this.addSpinner(button);
+        this.cleanTurnstile();
+        this.addTurnstile(this.action)?.insertBefore(button);
+    },
+};
+
+publicWidget.registry.s_website_form.include({
+    ...turnStile,
+
+    /**
+     * @override
+     */
+    start: function () {
+        const res = this._super(...arguments);
+        if(session.turnstile_site_key) {
+            let button = this.el.querySelector('.s_website_form_send, .o_website_form_send');
+            this.addSpinner(button);
```

Comment:
Complete crash on the contact us button
![image](https://github.com/user-attachments/assets/64046332-047d-4f47-8f84-dd110939f6ae)

It looks like you didn't test your dev throughfully :D